pkg/controller/istiocontrolplane/pruningdetails.go

```diff
@@ -104,7 +104,11 @@ var (
 
 // NewPruningDetails creates a new PruningDetails object specific to the instance.
 func NewIstioPruningDetails(instance *v1alpha2.IstioControlPlane) helmreconciler.PruningDetails {
-	gvk := instance.GetObjectKind().GroupVersionKind()
+	gvk := schema.GroupVersionKind{
+		Version: "v1alpha2",
+		Group:   "install.istio.io",
+		Kind:    "IstioControlPlane",
+	}
```

Comment:
Please define these as consts in a common package somewhere. I think Group is already defined...